### PR TITLE
nuraft: init at 1.1.2

### PR DESCRIPTION
**Pull request by wheelsandmetal**
###### Motivation for this change
Add nuraft lib

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/nuraft/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ boost172 asio openssl zlib ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/eBay/NuRaft";
+    description = "C++ implementation of Raft core logic as a replication library ";
+    license = licenses.asl20;
+  };
```
Comment (AndersonTorres) [inline pkgs/development/libraries/nuraft/default.nix:19]:
```suggestion
  platforms = platforms.linux;
  };
```

File: pkgs/development/libraries/nuraft/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    url = "https://github.com/eBay/NuRaft/archive/v${version}.tar.gz";
+    sha256 = "sha256-VZT2q32Md48IkOmtj52P3fsF3jkW60WjFqoglITjHVM=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ boost172 asio openssl zlib ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/eBay/NuRaft";
+    description = "C++ implementation of Raft core logic as a replication library ";
```
Comment (AndersonTorres) [inline pkgs/development/libraries/nuraft/default.nix:17]: Trailing whitespace.

```suggestion
    description = "C++ implementation of Raft core logic as a replication library";
```

**Comment by wheelsandmetal** [comment]
Not sure who to set as maintainer

**Comment by AndersonTorres** [comment]
@wheelsandmetal 

If you are interested in maintain this package, you can add yourself at `maintainers/maintainer-list.nix` and add it to meta.maintainers list, or else just keep the list empty (basically "orphaning" the package).

Also, squash your commits to more meaningful messages. We don't pollut the commit log with small messages as "applying suggested changes" or "fix trailing whitespace". Just the original message "nuraft: init at 1.1.2" suffices.

**Comment by wheelsandmetal** [comment]
> @wheelsandmetal
> 
> If you are interested in maintain this package, you can add yourself at `maintainers/maintainer-list.nix` and add it to meta.maintainers list, or else just keep the list empty (basically "orphaning" the package).

Sure

> Also, squash your commits to more meaningful messages. We don't pollut the commit log with small messages as "applying suggested changes" or "fix trailing whitespace". Just the original message "nuraft: init at 1.1.2" suffices.

ah didn't know, thanks!

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
@wheelsandmetal

> > Also, squash your commits to more meaningful messages. We don't pollut the commit log with small messages as "applying suggested changes" or "fix trailing whitespace". Just the original message "nuraft: init at 1.1.2" suffices.
> 
> ah didn't know, thanks!

Split the commits:
- One for "add myself to maintainers"
- The next one "nuraft: init at 1.1.2"

**Review by AndersonTorres — APPROVED** [review]
LGTM, LGTBorg

**Pull request merged**